### PR TITLE
Disable builder blob cache

### PR DESCRIPTION
**Pull request by adambkaplan**
Address flakes - nodes running out of disk space.

### CODE REVIEW

**Comment by adambkaplan** [comment]
/assign @bparees 

fyi @nalind

**Comment by nalind** [comment]
LGTM

**Comment by bparees** [comment]
@adambkaplan you broke the unit tests :)

**Comment by bparees** [comment]
feel free to re-lgtm yourself after you fix them

File: pkg/build/controller/strategy/util.go [hunk 1]
```diff
@@ -404,6 +404,9 @@ func updateConfigsForContainer(c corev1.Container, volumeName string, configDir
 	c.Env = append(c.Env, corev1.EnvVar{Name: "BUILD_REGISTRIES_DIR_PATH", Value: registriesDirPath})
 	c.Env = append(c.Env, corev1.EnvVar{Name: "BUILD_SIGNATURE_POLICY_PATH", Value: signaturePolicyPath})
 	c.Env = append(c.Env, corev1.EnvVar{Name: "BUILD_STORAGE_CONF_PATH", Value: storageConfPath})
+	// Disable blob cache - worker nodes running out of disk
+	// Set BUILD_BLOBCACHE_DIR to empty value per https://github.com/openshift/builder/blob/master/pkg/build/builder/cmd/builder.go#L139
+	c.Env = append(c.Env, corev1.EnvVar{Name: "BUILD_BLOBCACHE_DIR", Value: ""})
```
Comment (smarterclayton) [inline pkg/build/controller/strategy/util.go]: This seems like something the operator should tune based on the size of disks on nodes.  On nodes with small disks the blob cache may not be worth it by default.  On nodes with large disks it's probably always worth it.

Comment (adambkaplan) [reply pkg/build/controller/strategy/util.go]: I can see this being a challenge to do automatically if admins use node selectors to push builds onto specific nodes and/or use multiple machine configs.

**Comment by bparees** [comment]
@nalind @adambkaplan 
I am starting to suspect that disabling the blobcache is somehow breaking the layers test, as it is failing pretty consistently only in this PR:

```
fail [github.com/openshift/origin/test/extended/images/layers.go:205]: Expected
    <string>: sha256:57c14dd66db0390dbf6da578421c077f6de8e88edd0815b4caa94607ba5f4c09
to equal
    <string>: sha256:cfa9e50250a5295816f2f040d09cc596e01bae86844a34da924d2826bedd12b7
```

is that possible?

**Comment by bparees** [comment]
/hold

until @nalind and @adambkaplan can look at/comment on the layer cache failure.

**Comment by bparees** [comment]
@adambkaplan we're back to passing w/ the blob cache enabled (using emptydir storage) right?  so we can close this?

**Comment by adambkaplan** [comment]
Closing - we are keeping the blob cache (and moving it to use an `EmptyDir` volume in openshift/origin#21802).

**Pull request closed without merging**